ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -69,7 +71,89 @@ private[server] object ServerHelpers {
         .evalTap(e => ready.complete(e.map(_._1.toInetSocketAddress)))
         .rethrow
         .flatMap(_._2)
+    serverInternal(
+      server,
+      httpApp: HttpApp[F],
+      tlsInfoOpt: Option[(TLSContext[F], TLSParameters)],
+      shutdown: Shutdown[F],
+      // Defaults
+      errorHandler: Throwable => F[Response[F]],
+      onWriteFailure: (Option[Request[F]], Response[F], Throwable) => F[Unit],
+      maxConnections: Int,
+      receiveBufferSize: Int,
+      maxHeaderSize: Int,
+      requestHeaderReceiveTimeout: Duration,
+      idleTimeout: Duration,
+      logger: Logger[F],
+      true
+    )
+  }
+
+  def unixSocketServer[F[_]: Async](
+      unixSockets: UnixSockets[F],
+      unixSocketAddress: UnixSocketAddress,
+      deleteIfExists: Boolean,
+      deleteOnClose: Boolean,
+      httpApp: HttpApp[F],
+      tlsInfoOpt: Option[(TLSContext[F], TLSParameters)],
+      ready: Deferred[F, Either[Throwable, InetSocketAddress]],
+      shutdown: Shutdown[F],
+      // Defaults
+      errorHandler: Throwable => F[Response[F]],
+      onWriteFailure: (Option[Request[F]], Response[F], Throwable) => F[Unit],
+      maxConnections: Int,
+      receiveBufferSize: Int,
+      maxHeaderSize: Int,
+      requestHeaderReceiveTimeout: Duration,
+      idleTimeout: Duration,
+      logger: Logger[F]
+  ) = {
+    val server =
+      // Our interface has an issue
+      Stream
+        .eval(
+          ready.complete( // This is a lie, there isn't any signal from fs2 when the server is actually ready
```

Comment:
Should we call this after `unixSockets.server` has returned ?

Reply:
`unixSockets.server` returns a stream of sockets so there's not much wiggle room there I think.

Reply:
in the past, this kind of behavior has led to poor UX and flaky test suites since we don't know when we are cleared to make a request to the server. we could instrument retries but that isn't a silver bullet and it also pollutes the test suites

Reply:
We agree in discord we need to look into the fs2 interface at some point. But that this is alright for where we are at now.